keps/sig-release/20190316-rebase-images-to-distroless.md

```diff
@@ -0,0 +1,149 @@
+### Distroless and Previous Work
+"Distroless images contain only your application and its runtime dependencies. They do not contain package managers, shells or any other programs you would expect to find in a standard Linux distribution.”  (from [github.com/GoogleContainerTools/distroless/README](https://github.com/GoogleContainerTools/distroless))
+Distroless supports the dependencies where “FROM scratch” misses and more light-weighted than debian or alpine. Meanwhile, distroless “improves the signal to noise of scanners (e.g. CVE) and reduces the burden of establishing provenance to just what you need.”(from [github.com/GoogleContainerTools/distroless/README](https://github.com/GoogleContainerTools/distroless))
+
+Using Distroless as a common image base is originally proposed as an exploration area in [the base image policy for system containers](https://github.com/kubernetes/kubernetes/issues/40248). Tim(tallclair@) has driven the effort on defining and establishing the base image policy (main changes):
+
+* Add Alpine iptable as base image for kube-proxy. Previously kube-proxy is based on debian iptable image.  (This direction is scrapped, see [issue/39696](https://github.com/kubernetes/kubernetes/issues/39696) for details)
+Rebase busybox images to debian-base.
+* Rebase busybox images to debian-base.
+* Rebase certain alpine images to debian-base
+* Image related works on migrating apiserver-builder to kube-builder
+
+The distroless solution is filed separately in [issue/70249](https://github.com/kubernetes/kubernetes/issues/70249). The doc, as a more up-to-date version, is slightly different than the original issue.
+
+## Proposal
+
+The approaches to rebase different containers vary significantly due to the technical implementations, development plans or even the legacy reasons. For example, [ingress-gce/fuzzer](https://github.com/kubernetes/ingress-gce/blob/64eee7e3521680057b071d5e9bebaa215086a4bc/Dockerfile.fuzzer) was based on alpine and can't be switched to distroless directly due to the fact that it needs shell to redirect the glog file. To allow the images to be based on distroless (which doesn't contain shell), we firstly need to remove the dependency on the shell (use klog instead of glog), and then rebase the image. (Related PR [pr/682](https://github.com/kubernetes/ingress-gce/pull/682]), [pr/666](https://github.com/kubernetes/ingress-gce/pull/666))
+There are 4 main efforts on this KEP.
+
+1. The images are built via bazel. In such case, we will update the bazel BUILD rule to switch to base from distroless image. This method applies for the core containers like kube-apiserver, kube-controller-manager, cloud-controller-manager, kube-scheduler. As for kube-proxy, it is based on debian-iptable which distroless doesn't support iptables yet. Thus, kube-proxy won't be changed.
```

Comment:
nit: grammar
```suggestion
1. The images are built via bazel. In such case, we will update the bazel BUILD rule to switch to the base image to distroless/static. This method applies for the core containers like kube-apiserver, kube-controller-manager, cloud-controller-manager, kube-scheduler. As for kube-proxy, it is based on debian-iptable which distroless doesn't support iptables yet. Thus, kube-proxy won't be changed.
```

Also, I think you should be clear that you're suggesting distroless/static everywhere. Optionally, you could state in the beginning of the document that you use distroless to refer to gcr.io/distroless/static everywhere.

Reply:
Sure. Updated everywhere.